File: erp-app/src/app/modules/accounting/models/account-statement-dto.ts
import { AccountTypeEnum } from "src/app/shared/enums/account-type.enum";
import { BillTypeEnum } from "src/app/shared/enums/bill-type.enum";
import { PaymentMethodEnum } from "src/app/shared/enums/payment-method.enum";
import { TransactionTypeEnum } from "src/app/shared/enums/transaction-type.enum";
import { BaseDTO } from "src/app/shared/models/base-dto.model";

export class AccountStatementDTO extends BaseDTO {

    number: number;
    date: string | undefined;
    clientVendorId: number | null;
    beneficiaryName: string | undefined;
    //transactionTypeId: TransactionTypeEnum;
    paymentMethodId: PaymentMethodEnum;
    debit: number = 0;
    credit: number = 0;
    refNo: string;
    notes: string;
    isBilled: boolean;
    billId: number;
    billType: BillTypeEnum;
}

File: ERPServer/Shared/Shared/Entities/Accouting/AccountStatementDTO.cs
using Shared.Entities.Shared;
using Shared.Enums;

namespace Shared.Entities.Accouting
{
    public class AccountStatementDTO : BaseDTO
    {
        public string Number { get; set; }
        public string Date { get; set; }
        public long? ClientVendorId { get; set; }
        public string BeneficiaryName { get; set; }
        public PaymentMethodEnum? PaymentMethodId { get; set; }
        public decimal Debit { get; set; }
        public decimal Credit { get; set; }
        public string RefNo { get; set; }
        public string Notes { get; set; }
        public bool IsCancel { get; set; }
        public bool IsBilled { get; set; }
        public long? BillId { get; set; }
        public long? RepresentiveId { get; set; }
        public BillTypeEnum? BillType { get; set; }

    }
}
